File: Procons.Durrah.Api/ClientApp/app/Models/Worker.ts
import { SearchCriteriaParams, NameValuePair } from './ApiRequestType';


export class Worker {
    constructor(public workerName: string, public name: string,
        public birthDate: string,
        public gender: string,
        public nationality: string,
        public religion: string,
        public maritalStatus: string,
        public language: string,
        public languages: NameValuePair[],
        public photo: string,
        public weight: string,
        public height: string,
        public education: string,
        public video: string,
        public passportNumber: string,
        public passportIssDate: string,
        public passportPoIssue: string,
        public passportExpDate: string,
        public civilId: string,
        public yearsOfExperience:number,

        public serialNumber: string,
        public agent: string,
        public code: string,
        public workerCode: string,
        public passport: string,
        public license: string,
        public status: string,
        public passportCopy: any,

        public salary: number,
        public price: number,
        public workerType: any,
        public mobile: string,

        public hobbies: string,
        public location: string,
        public isNew: string,
        public period: boolean,
        public experiences?: Experience[]
    ) { }
}

export class WorkerManagementData {
    searchCriteria: SearchCriteriaParams;
    workerServerData: Worker[];
    workerDisplayData: Worker[];
    constructor() { }
}

export class Experience {
    title = ''
    description = ''
    companyName = ''
    country = ''
    location = ''
    startDate = ''
    endDate = ''
}

//Problem is that some views expect worker keys and othe expect worker values ? 
// a- create a function that can  convert from one state to the other
// b- have a new interface that contains all the different logic
// c- consider how much code needs to be changed ...

File: Procons.Durrah.Common/Models/Worker.cs
using System.Collections.Generic;

namespace Procons.Durrah.Common
{
    public class Worker
    {
        /// <summary>
        /// Worker Code
        /// </summary>
        public string WorkerCode { get; set; }
        public string WorkerName { get; set; }
        public string SerialNumber { get; set; }
        public string Agent { get; set; }
        public string Mobile { get; set; }
        public int Age { get; set; }
        /// <summary>
        /// Related Item Name
        /// </summary>
        public string Name { get; set; }
        /// <summary>
        /// Related Item Code
        /// </summary>
        public string Code { get; set; }
        public string BirthDate { get; set; }
        public string Gender { get; set; }
        public string Nationality { get; set; }
        public string Religion { get; set; }
        public string MaritalStatus { get; set; }
        public string Language { get; set; }
        public string Photo { get; set; }
        public string License { get; set; }
        public double Price { get; set; }
        public double Salary { get; set; }
        public string Weight { get; set; }
        public string Height { get; set; }
        public string Education { get; set; }
        public string Passport { get; set; }
        public string Video { get; set; }
        public string PassportNumber { get; set; }
        public string PassportIssDate { get; set; }
        public string PassportExpDate { get; set; }
        public string PassportPoIssue { get; set; }
        public string CivilId { get; set; }
        public string Status { get; set; }
        public string WorkerType { get; set; }
        public List<LookupItem> Languages { get; set; }

    }
}